setup.py

```diff
@@ -135,97 +115,26 @@ def run(self):
-    pass
-
-if WHEEL_INSTALLED:
-    class bdist_wheel_command(_command_template, bdist_wheel):
-        pass
-
-
-class cmake_extension(Extension):
-    def __init__(self, name):
-        Extension.__init__(self, name, sources=[])
-
-
-class cmake_build(build_ext):
-    def run(self):
-        if not check_cmake():
-            raise RuntimeError('CMake is not available. CMake 3.12 is required.')
-
-        # The path where CMake will be configured and Arbor will be built.
-        build_directory = os.path.abspath(self.build_temp)
-        # The path where the package will be copied after building.
-        lib_directory = os.path.abspath(self.build_lib)
-        # The path where the Python package will be compiled.
-        source_path = build_directory + '/python/arbor'
-        # Where to copy the package after it is built, so that whatever the next phase is
-        # can copy it into the target 'prefix' path.
-        dest_path = lib_directory + '/arbor'
-
-        opt = cl_opt()
-        cmake_args = [
-            '-DARB_WITH_PYTHON=on',
-            '-DPYTHON_EXECUTABLE=' + sys.executable,
-            '-DARB_WITH_MPI={}'.format( 'on' if opt['mpi'] else 'off'),
-            '-DARB_VECTORIZE={}'.format('on' if opt['vec'] else 'off'),
-            '-DARB_ARCH={}'.format(opt['arch']),
-            '-DARB_GPU={}'.format(opt['gpu']),
-            '-DARB_WITH_NEUROML={}'.format( 'on' if opt['neuroml'] else 'off'),
-            '-DARB_USE_BUNDLED_LIBS={}'.format('on' if opt['bundled'] else 'off'),
-            '-DCMAKE_BUILD_TYPE=Release' # we compile with debug symbols in release mode.
-        ]
-
-        print('-'*5, 'command line arguments: {}'.format(opt))
-        print('-'*5, 'cmake arguments: {}'.format(cmake_args))
-
-        build_args = ['--config', 'Release']
-
-        # Assuming Makefiles
-        build_args += ['--', f'-j{opt["makejobs"]}']
-
-        env = os.environ.copy()
-        env['CXXFLAGS'] = '{}'.format(env.get('CXXFLAGS', ''))
-        if not os.path.exists(self.build_temp):
-            os.makedirs(self.build_temp)
-
-        # CMakeLists.txt is in the same directory as this setup.py file
-        cmake_list_dir = os.path.abspath(os.path.dirname(__file__))
-        print('-'*20, 'Configure CMake')
-        subprocess.check_call(['cmake', cmake_list_dir] + cmake_args,
-                              cwd=self.build_temp, env=env)
-
-        print('-'*20, 'Build')
-        cmake_cmd = ['cmake', '--build', '.'] + build_args
-        subprocess.check_call(cmake_cmd,
-                              cwd=self.build_temp)
-
-        # Copy from build path to some other place from whence it will later be installed.
-        # ... or something like that
-        # ... setuptools is an enigma monkey patched on a mystery
-        if not os.path.exists(dest_path):
-            os.makedirs(dest_path, exist_ok=True)
-        self.copy_tree(source_path, dest_path)
-
-setuptools.setup(
+opt = cl_opt()
+setup(
```

Comment:
So I still don't see a `packages=["arbor"]` or similar here. Is this done automatically by `scikit-build` or is the CMake build still producing the `__init__.py` file and being picked up together with `arbor*.so`?

Reply:
Although I am not sure yet, I think this `setup.py` contains leftovers that are no longer used. I'm having trouble getting out-of-(source-)tree builds to work, and it looks like we might be relying on many automatic things where we probably shouldn't. Rabbit hole: https://github.com/pypa/pip/issues/10573#issuecomment-940895948

I'm getting a headache trying to build a mental image of what scikit is doing, and why it fails for out-of-tree builds (which a `bdist_wheel` build is, or at least until pip 21.3, but see rabbit hole above for the many components possibly interfering).